snuba/state/rate_limit.py

```diff
@@ -124,16 +129,27 @@ def rate_limit(
     query_id = str(uuid.uuid4())
 
     now = time.time()
-    bypass_rate_limit, rate_history_s = state.get_configs(
-        [("bypass_rate_limit", 0), ("rate_history_sec", 3600)]
-        #                               ^ number of seconds the timestamps are kept
+    bypass_rate_limit, rate_history_s, v2_cluster_rollout_rate = state.get_configs(
+        [
+            ("bypass_rate_limit", 0),
+            # number of seconds the timestamps are kept
+            ("rate_history_sec", 3600),
+            ("rate_limit_use_v2_cluster", 0.0),
+        ]
     )
     assert isinstance(rate_history_s, (int, float))
+    assert isinstance(v2_cluster_rollout_rate, float)
 
     if bypass_rate_limit == 1:
         yield None
         return
 
+    should_use_v2 = (
```

Comment:
I think this sharding choice logic should be inside the RedisCache, instead of duplicated in the code every place we want a v2 shard. Maybe we need a wrapper class around multiple `RedisCache`?

Reply:
i agree with this but i had hoped this code will be removed after we have moved the usecases over (so, within one week)

Reply:
"shard" is a misnomer i guess, we shard the keyspace across old/new cluster and make one shard gradually bigger in order to migrate traffic/load